### ISSUE TITLE
[ts][webgl][widget] webgl busted in latest Safari (10.1)

### ISSUE THREAD

**Issue by geddski** [EsotericSoftware/spine-runtimes#886]
While testing my app in the latest Safari (v10.1 released in March) I noticed that none of my animations show up at all. No errors or anything. 

None of the [webgl demos listed here](https://github.com/EsotericSoftware/spine-runtimes/tree/master/spine-ts) work either.

I'm about to release my app and would love for it to work for people using Safari.

**Comment by geddski** [comment]
Specifically I'm using Widget runtime

**Comment by badlogic** [comment]
Bah, Safari. I take a look!

**Comment by badlogic** [comment]
![https://libgdx.badlogicgames.com/uploads/Screen%20Shot%202017-04-26%20at%201.44.06%20PM-Rek2KzqVnd.png](https://libgdx.badlogicgames.com/uploads/Screen%20Shot%202017-04-26%20at%201.44.06%20PM-Rek2KzqVnd.png)

Top left: Chome, Right: Firefox, Bottom: Safari (10.0.3).

Will update to latest Safari and see what happens there. I get zero warnings in any browser.

**Comment by badlogic** [comment]
Simplest repro:

```Javascript
<html>
<script src="../../build/spine-webgl.js"></script>
<script src="https://code.jquery.com/jquery-3.1.0.min.js"></script>
<style>
	* { margin: 0; padding: 0; }
	body, html { height: 100% }
	canvas { position: absolute; width: 100% ;height: 100%; }
</style>
<body>
<div id="label" style="position: absolute; top: 0; left: 0; color: #fff; z-index: 10"></div>
<canvas id="canvas"></canvas>
</body>
<script>
var canvas = document.getElementById("canvas");
canvas.width = canvas.clientWidth;
canvas.height = canvas.clientHeight;
var context = new spine.webgl.ManagedWebGLRenderingContext(canvas);
var gl = context.gl;

var batcher = new spine.webgl.PolygonBatcher(context);
var shader = spine.webgl.Shader.newTwoColoredTextured(context);
var img = new Image();
var texture = null;
img.src = "assets/spineboy.png";
img.onload = function (ev) {
	texture = new spine.webgl.GLTexture(context, img);
};
var camMatrix = new spine.webgl.Matrix4();
var vertices = [ -1, -1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0,
				 1, -1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0,
				 0, 1, 1, 1, 1, 1, 0.5, 1, 0, 0, 0, 0];
var indices = [ 0, 1, 2 ];

requestAnimationFrame(render);

function render() {
	gl.clearColor(1, 0, 0, 1);
	gl.clear(gl.COLOR_BUFFER_BIT);

	shader.bind();
	shader.setUniform4x4f(spine.webgl.Shader.MVP_MATRIX, camMatrix.values);
	batcher.begin(shader);
	batcher.draw(texture, vertices, indices);
	batcher.end();
	shader.unbind();

	requestAnimationFrame(render);
}
</script>
```
Renders white triangle on Safari, renders correctly textured triangle everywhere else.

**Comment by badlogic** [comment]
The problem seems to be blending. If i disable blending, things get rendered properly (without blending of course...).

**Comment by badlogic** [comment]
The issue was that we used WebGL constants like `SRC_ALPHA` via `WebGLRenderingContext.SRC_ALPHA`. These are defined in Edge, Firefox and Chrome, but it seems Safari no longer defines them in that places. One needs an instance of `WebGLRenderingContext` to get to those constants. Fixed via https://github.com/EsotericSoftware/spine-runtimes/commit/6a5e035cb5bf38404da9b26234d067e366da99a4, tested in Safari 10.0.3, updating to latest now and testing as well.
